File: VB/Applications/WinFormsSerilogVBNoDI/Helpers/LoggingHelper.vb
Imports LogViewerVB.Core
Imports Microsoft.Extensions.Configuration
Imports Microsoft.Extensions.Logging
Imports Serilog
Imports Serilog.Sinks.LogViewVB.Core
Imports WinFormsSerilogVBNoDI.DataStores

Namespace Helpers

    ' application-wide DataStoreLogger Factory ... returns a wired up Logger instance
    Public Module LoggingHelper

#Region "Constructors"

        Sub New()

            Dim configuration As IConfigurationRoot = New ConfigurationBuilder() _
                .Initialize() _
                .Build()

            'Log.Logger = New LoggerConfiguration() _
            '.ReadFrom.Configuration(configuration) _
            '.WriteTo.DataStoreLoggerSink(Function() MainControlsDataStore.DataStore) _
            '.CreateLogger()

            Log.Logger = New LoggerConfiguration() _
                    .ReadFrom.Configuration(configuration) _
                    .WriteTo.DataStoreLoggerSink(
                        Function() MainControlsDataStore.DataStore,
                        Sub(options)

                            options.Colors(LogLevel.Trace) = New LogEntryColor() With
                            {
                                .Foreground = Color.White,
                                .Background = Color.DarkGray
                            }

                            options.Colors(LogLevel.Debug) = New LogEntryColor() With
                            {
                                .Foreground = Color.White,
                                .Background = Color.Gray
                            }

                            options.Colors(LogLevel.Information) = New LogEntryColor() With
                            {
                                .Foreground = Color.White,
                                .Background = Color.DodgerBlue
                            }

                            options.Colors(LogLevel.Warning) = New LogEntryColor() With
                            {
                                .Foreground = Color.White,
                                .Background = Color.Orchid
                            }

                        End Sub) _
                    .CreateLogger()

            ' wire up the loggers
            Factory = LoggerFactory.Create(Sub(LoggingBuilder) LoggingBuilder.AddSerilog(Log.Logger))

        End Sub

#End Region

#Region "Properties"

        Public ReadOnly Property Factory As ILoggerFactory

#End Region

#Region "Methods"

        Friend Sub CloseAndFlush()

            Log.CloseAndFlush()

        End Sub

#End Region

    End Module

End Namespace

File: CSharp/Applications/AvaloniaSerilogNoDI/Helpers/LoggingHelper.cs
using AvaloniaSerilogNoDI.DataStores;
using Common.Core.Extensions;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;
using Serilog;
using Serilog.Sinks.LogView.Core;
using System.Drawing;

namespace AvaloniaSerilogNoDI.Helpers;

// application-wide DataStoreLogger Factory ... returns a wired up Logger instance
public static class LoggingHelper
{
    #region Constructors

    static LoggingHelper()
    {
        IConfigurationRoot configuration = new ConfigurationBuilder()
            .Initialize()
            .Build();

        Log.Logger = new LoggerConfiguration()
            .ReadFrom.Configuration(configuration)
            .WriteTo.DataStoreLoggerSink(
                //dataStoreProvider: () => MainControlsDataStore.DataStore

                dataStoreProvider: () => MainControlsDataStore.DataStore,
                options =>
                {
                    options.Colors[LogLevel.Trace] = new()
                    {
                        Foreground = Color.White,
                        Background = Color.DarkGray
                    };

                    options.Colors[LogLevel.Debug] = new()
                    {
                        Foreground = Color.White,
                        Background = Color.Gray
                    };

                    options.Colors[LogLevel.Information] = new()
                    {
                        Foreground = Color.White,
                        Background = Color.DodgerBlue
                    };

                    options.Colors[LogLevel.Warning] = new()
                    {
                        Foreground = Color.White,
                        Background = Color.Orchid
                    };
                }
            )
            .CreateLogger();

        // wire up the loggers
        Factory = LoggerFactory.Create(loggingBuilder => loggingBuilder.AddSerilog(Log.Logger));
    }

    #endregion

    #region Properties

    public static ILoggerFactory Factory { get; }

    #endregion

    #region Methods

    public static void CloseAndFlush()
        => Log.CloseAndFlush();

    #endregion
}